src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -291,6 +304,12 @@ PyChipError pychip_DeviceController_StackInit(Controller::Python::StorageAdapter
     //
     DeviceControllerFactory::GetInstance().RetainSystemState();
 
+    auto engine = chip::app::InteractionModelEngine::GetInstance();
+    PyReturnErrorOnFailure(ToPyChipError(PyChipCheckInDelegate::GetInstance().Init(&sICDClientStorage, engine)));
```

Comment:
The interaction model engine is too big an interface for check-in delegate registration. Why is the whole InteractionModelEngine needed?

Reply:
it is used to re-trigger the inactive subscription(mpImEngine->OnActiveModeNotification(mICDClientInfo.peer_node);) via DefaultCheckInDelegate/RefreshKeySender, we need look for the right readClient for subscription based upon the mpActiveReadClientList, this list is coupled with IM engine at this moment. 

void InteractionModelEngine::OnActiveModeNotification(ScopedNodeId aPeer)
{
    for (ReadClient * pListItem = mpActiveReadClientList; pListItem != nullptr;)
    {
        auto pNextItem = pListItem->GetNextClient();
        // It is possible that pListItem is destroyed by the app in OnActiveModeNotification.
        // Get the next item before invoking `OnActiveModeNotification`.
        if (ScopedNodeId(pListItem->GetPeerNodeId(), pListItem->GetFabricIndex()) == aPeer)
        {
            pListItem->OnActiveModeNotification();
        }
        pListItem = pNextItem;
    }
}

https://github.com/project-chip/connectedhomeip/blob/master/src/app/icd/client/DefaultCheckInDelegate.cpp#L58
https://github.com/project-chip/connectedhomeip/blob/master/src/app/icd/client/RefreshKeySender.cpp#L67